test/index.js

```diff
@@ -545,6 +545,109 @@ describe('def', function() {
                    'as its first argument; received ["abc"]'));
   });
 
+  it('supports the "ValidNumber" type', function() {
+    var def = $.create($.env.concat([$.ValidNumber]));
+
+    // add :: ValidNumber -> ValidNumber
+    var inc = def('inc',
+                  {},
+                  [$.ValidNumber, $.ValidNumber],
+                  function(x) { return x + 1; });
+
+    throws(function() { inc(NaN); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type ValidNumber ' +
+                   'as its first argument; received NaN'));
+
+    eq(inc(1), 2);
+  });
+
+  it('supports the "FiniteNumber" type', function() {
+    var def = $.create($.env.concat([$.FiniteNumber]));
+
+    // add :: FiniteNumber -> FiniteNumber
+    var inc = def('inc',
+                  {},
+                  [$.FiniteNumber, $.FiniteNumber],
+                  function(x) { return x + 1; });
+
+    throws(function() { inc(Infinity); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type FiniteNumber ' +
+                   'as its first argument; received Infinity'));
+
+    throws(function() { inc(-Infinity); },
+           errorEq(TypeError,
+                   '‘inc’ expected a value of type FiniteNumber ' +
+                   'as its first argument; received -Infinity'));
+
+    eq(inc(-1), 0);
+  });
+
+  it('supports the "NonZeroValidNumber" type', function() {
+    var def = $.create($.env.concat([$.ValidNumber, $.NonZeroValidNumber]));
+
+    // div :: ValidNumber -> NonZeroValidNumber -> ValidNumber
+    var div = def('div',
+                  {},
+                  [$.ValidNumber, $.NonZeroValidNumber, $.ValidNumber],
+                  function(a, b) { return a / b; });
+
+    throws(function() { div(3, 0); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroValidNumber ' +
+                   'as its second argument; received 0'));
+
+    throws(function() { div(3, Number(0)); },
+      errorEq(TypeError,
+        '‘div’ expected a value of type NonZeroValidNumber ' +
+        'as its second argument; received 0'));
+
+    throws(function() { div(3, -0); },
+           errorEq(TypeError,
+                   '‘div’ expected a value of type NonZeroValidNumber ' +
+                   'as its second argument; received -0'));
+
+    throws(function() { div(3, Number(-0)); },
+      errorEq(TypeError,
+        '‘div’ expected a value of type NonZeroValidNumber ' +
+        'as its second argument; received -0'));
+
+    eq(div(4, 2), 2);
+  });
+
+  it('supports the "NonZeroFiniteNumber" type', function() {
+    var def = $.create($.env.concat([$.ValidNumber, $.NonZeroFiniteNumber]));
+
+    // div :: ValidNumber -> NonZeroFiniteNumber -> ValidNumber
+    var div = def('div',
+      {},
+      [$.ValidNumber, $.NonZeroFiniteNumber, $.ValidNumber],
+      function(a, b) { return a / b; });
```

Comment:
The good news is that I've found the setting that fixes this, I think. We shall see...